src/utils/DepositUtils.ts

```diff
@@ -82,7 +103,18 @@ export async function queryHistoricalDepositForFill(
     }
   }
 
-  return validateFillForDeposit(fill, deposit) ? deposit : undefined;
+  if (validateFillForDeposit(fill, deposit)) {
```

Comment:
Passing back the reasons is really great! Not necessary for this PR, but we may want to also deal with errors in the findDeposit call similarly, so they can be attributed with reasons instead of just throwing an error in the middle of this function.